pkg/models/vulnerability.go

```diff
@@ -24,78 +24,79 @@ type Package struct {
 //
 // See: https://ossf.github.io/osv-schema/#affectedrangesevents-fields
 type Event struct {
-	Introduced   string `json:"introduced,omitempty"`
-	Fixed        string `json:"fixed,omitempty"`
-	LastAffected string `json:"last_affected,omitempty"`
-	Limit        string `json:"limit,omitempty"`
+	Introduced   string `json:"introduced,omitempty" yaml:"introduced,omitempty"`
+	Fixed        string `json:"fixed,omitempty" yaml:"fixed,omitempty"`
+	LastAffected string `json:"last_affected,omitempty" yaml:"last_affected,omitempty"`
+	Limit        string `json:"limit,omitempty" yaml:"limit,omitempty"`
 }
 
 // Range describes the affected range of given version for a specific package.
 //
 // See: https://ossf.github.io/osv-schema/#affectedranges-field
 type Range struct {
-	Type             RangeType              `json:"type"`
-	Events           []Event                `json:"events"`
-	DatabaseSpecific map[string]interface{} `json:"database_specific,omitempty"`
+	Type             RangeType              `json:"type" yaml:"type"`
+	Events           []Event                `json:"events" yaml:"events"`
+	Repo             string                 `json:"repo,omitempty" yaml:"repo,omitempty"`
+	DatabaseSpecific map[string]interface{} `json:"database_specific,omitempty" yaml:"database_specific,omitempty"`
 }
 
 // Severity is used to describe the severity of a vulnerability for an affected
 // package using one or more quantitative scoring methods.
 //
 // See: https://ossf.github.io/osv-schema/#severity-field
 type Severity struct {
-	Type  SeverityType `json:"type"`
-	Score string       `json:"score"`
+	Type  SeverityType `json:"type" yaml:"type"`
+	Score string       `json:"score" yaml:"score"`
 }
 
 // Affected describes an affected package version, meaning one instance that
 // contains the vulnerability.
 //
 // See: https://ossf.github.io/osv-schema/#affected-fields
 type Affected struct {
-	Package           Package                `json:"package"`
-	Severity          []Severity             `json:"severity,omitempty"`
-	Ranges            []Range                `json:"ranges,omitempty"`
-	Versions          []string               `json:"versions,omitempty"`
-	DatabaseSpecific  map[string]interface{} `json:"database_specific,omitempty"`
-	EcosystemSpecific map[string]interface{} `json:"ecosystem_specific,omitempty"`
+	Package           *Package               `json:"package,omitempty" yaml:"package,omitempty"`
```

Comment:
Is this a required change? It may break things subtlely for some users expecting copy semantics instead?

Reply:
I can change it back. It does change the output to always include the package, even if it has been omitted.

The workaround for this would be to re-add `omitempty` to `name` and `ecosystem` in `Package`

Reply:
Public dependents appear to be only Guac and Scorecard.

Reply:
I guess the question is whether this is important to get right early even if a few dependents break, or whether it should be done later with a v2 release?

Reply:
Let's just maintain compatibility for now, and re-add omitempty to `name` and `ecosystem`.

Reply:
An alternative would be to add a custom MarshalYAML/MarshalJSON that uses reflection to remove the field.

Reply:
Probably a bit of an overkill at this point. Is there much benefit to not having `omitempty` on the `name` and `ecosystem` fields?

Reply:
Also I may have a bit of an misunderstanding about how this all works in Go here. If we just revert back to what we had before (with `omitempty` on the fields within the `Package`), would we just end up with:

```
"package": {}
```

Or would `package` itself be omitted too?